### ISSUE TITLE
[Feature] Merge package.json

### ISSUE THREAD

**Issue by wdfinch** [yarnpkg/berry#3290]
- [ ] I'd be willing to implement this feature ([contributing guide](https://yarnpkg.com/advanced/contributing))
- [ ] This feature is important to have in this repository; a contrib plugin wouldn't do

**Describe the user story**

One feature that would be cool would be the ability to merge package.json files into others. I have a large app with over 10 workspaces. In my package.json files, I have a lot of redundancy. I have settings like author and git repo that are always the same and a bunch of dependencies like prettier that I want in every project.

While constraints can enforce consistency, it seems like much less work to just write base package.json files and then compose larger package.json by selectively merging them in. So in other words rather than writing elaborate prolog code to say we made a mistake, instead we could just write simple package.json code that is correct once and import it to other package.json files. 

**Describe the solution you'd like**

In package.json files, perhaps there could be some field maybe within a yarn section where we can provide a path or array of paths to other package.json files. They could then be merged in that order and replace existing fields.
 
**Describe the drawbacks of your solution**

IDE sense might not work as well as it would not be aware of what packages you are using. Might be difficult to implement + more to maintain. Could break npm inference that relies on all the info being in package.json but not sure how much this matters for private packages and nested workspaces.

**Describe alternatives you've considered**

Right now I use `constraints`, `yarn up` and `yarn foreach` to bulk manage package.json. This isn't bad but sometimes I need to manually update 10+ package.json files and it's a pain. Additionally, I have these giant packages.json packages and it would be nice if they could be smaller as many fields are duplicated.

**Comment by arcanis** [comment]
One significant issue with that is that it would make it difficult/impossible to know exactly what affects the dependencies of a project. Right now you can inspect any `package.json` and you know exactly what it uses - no surprises. I'm quite fond of this property, and implicit inheritance like this doesn't seem worth losing it.

**Comment by wdfinch** [comment]
Wouldn't it not be possible to perform the merge operations first and then look at the resulting js object?

**Comment by arcanis** [comment]
My point is that a human operator looking at the `package.json` won't get a full understanding of which dependencies are inherited.

**Comment by wdfinch** [comment]
Makes sense. The current system is fine. I was just noticing that I have a lot of the same values in the package.json and was thinking of ways to simplify it or manage it from one place. Do you want to leave this open and see what other people think or close it?
